common/thanos/templates/_thanos_helpers.tpl

```diff
@@ -1,43 +1,77 @@
-{{/* Generated objectStorageConfigName */}}
-{{- define "thanos.objectStorageConfigName" -}}
-prometheus-{{- required ".Values.name" .Values.name -}}-thanos-storage-config
-{{- end -}}
-
 {{/* Thanos image. */}}
 {{- define "thanosimage" -}}
 {{- required ".Values.spec.baseImage missing" .Values.spec.baseImage -}}:{{- required ".Chart.appVersion missing" .Chart.AppVersion -}}
 {{- end -}}
 
 {{/* Thanos combined name */}}
-{{- define "thanos.fullName" -}}
-thanos-{{- required ".Values.name" .Values.name -}}
+{{- define "thanos.name" -}}
+{{- $name := index . 0 -}}
+{{- $root := index . 1 -}}
+{{- if and (not $root.Values.names) (not $root.Values.global.targets) (not $root.Values.name) -}}
```

Comment:
This is the same general check as for the new prometheus server chart. Basically, this helper is the first and the most used in the whole chart. It should ensure that at least one of the three mentioned values is filled, otherwise the chart should fail as expected.